pkg/plugins/builtin/email/email.go

```diff
@@ -70,38 +80,43 @@ func validConfig(config interface{}) error {
 func exec(stepName string, config interface{}, ctx interface{}) (interface{}, interface{}, error) {
 	cfg := config.(*Config)
 
-	parameters := struct {
-		From    string `json:"from"`
-		To      string `json:"to"`
-		Subject string `json:"subject"`
-		Body    string `json:"body"`
-	}{
-		cfg.From,
-		strings.Join(cfg.To, ","),
-		mime.BEncoding.Encode("UTF-8", cfg.Subject),
-		cfg.Body,
-	}
+	message := mail.NewMessage()
 
-	buffer := new(bytes.Buffer)
+	recipients := make([]string, len(cfg.To))
+	for i, recipient := range cfg.To {
+		recipients[i] = message.FormatAddress(recipient, "")
+	}
 
-	template := template.Must(template.New("emailTemplate").Parse(emailTemplate))
-	template.Execute(buffer, &parameters)
+	message.SetAddressHeader("From", cfg.FromAddress, cfg.FromName)
+	message.SetHeader("To", recipients...)
+	message.SetHeader("Subject", cfg.Subject)
+	message.SetBody(
+		http.DetectContentType([]byte(cfg.Body)),
+		cfg.Body,
+	)
 
-	auth := smtp.PlainAuth("", cfg.SMTPUsername, cfg.SMTPPassword, cfg.SMTPHostname)
+	port, _ := strconv.ParseUint(cfg.SMTPPort, 10, 64)
+	skipTLS, _ := strconv.ParseBool(cfg.SMTPSkipTLSVerify)
```

Comment:
ambiguous that this defaults to false when parsing "", WDYT? correct, but made me stop reading

Reply:
Do we *really* need a skipTLS parameter in 2019-2020 ?

Reply:
@nomonamo I added an explicit comment.
@Alkorin I think it's an usecase for testing purpose.

Reply:
I know it might seem obvious, but I'd mention what the default value is in case of err != nil, even if that's "not possible" per preemptive validation

```suggestion
port, _ := strconv.ParseUint(cfg.SMTPPort, 10, 64) // defaults to 0
skipTLS, _ := strconv.ParseBool(cfg.SMTPSkipTLSVerify) // defaults to false
```

which begs the question, false seems ok, but is 0 an appropriate default?

Reply:
I added default comments.
BTW port can't be not set default because https://golang.org/src/strconv/atoi.go?s=1789:1852#L61 generate an error in `validConfig` func (https://github.com/ovh/utask/pull/25/files#diff-7e310a8e891eb8f9cb73c9279c5a894bR51).